contracts/axelar-soroban-interfaces/src/types.rs

```diff
@@ -79,3 +90,79 @@ impl Proof {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use hex_literal::hex;
+    use soroban_sdk::{BytesN, Env, Vec};
+
+    use crate::types::{WeightedSigner, WeightedSigners};
+
+    #[test]
+    fn weighted_signers_hash() {
+        let env = Env::default();
+
+        let signers_array = [
+            WeightedSigner {
+                signer: BytesN::<32>::from_array(
+                    &env,
+                    &hex!("0a245a2a2a5e8ec439d1377579a08fc78ea55647ba6fcb1f5d8a360218e8a985"),
+                ),
+                weight: 3,
+            },
+            WeightedSigner {
+                signer: BytesN::<32>::from_array(
+                    &env,
+                    &hex!("0b422cf449d900f6f8eb97f62e35811c62eb75feb84dfccef44a5c1c3dbac2ad"),
+                ),
+                weight: 2,
+            },
+            WeightedSigner {
+                signer: BytesN::<32>::from_array(
+                    &env,
+                    &hex!("18c34bf01a11b5ba21ea11b1678f3035ef753f0bdb1d5014ec21037e8f99e2a2"),
+                ),
+                weight: 4,
+            },
+            WeightedSigner {
+                signer: BytesN::<32>::from_array(
+                    &env,
+                    &hex!("f683ca8a6d7fe55f25599bb64b01edcc5eeb85fe5b63d3a4f0b3c32405005518"),
+                ),
+                weight: 4,
+            },
+            WeightedSigner {
+                signer: BytesN::<32>::from_array(
+                    &env,
+                    &hex!("fbb4b870e800038f1379697fae3058938c59b696f38dd0fdf2659c0cf3a5b663"),
+                ),
+                weight: 2,
+            },
+        ];
+
+        let weighted_signers = WeightedSigners {
+            signers: Vec::from_array(&env, signers_array),
+            threshold: 8u128,
+            nonce: BytesN::<32>::from_array(
+                &env,
+                &hex!("8784bf7be5a9baaeea47e12d9e8ad0dec29afcbc3617d97f771e3c24fa945dce"),
+            ),
+        };
+
+        let hash = weighted_signers.hash(&env);
+        let signers_rotation_hash = weighted_signers.signers_rotation_hash(&env);
+
+        let expected_hash = BytesN::<32>::from_array(
+            &env,
+            &hex!("18b1ff823e202dab87cada621717e5be4955734bb973151eb489e6f1576ce3d4"),
```

Comment:
use golden files instead of hardcoding vectors in the test itself (using `goldie`)

Reply:
I should've left a comment here before your review. I already tried it, but the Goldie library requires using `std`, which gives an error. 

Error:
```
failed to resolve: could not find std in the list of imported crates
could not find std in the list of imported cratesrustcClick for full compiler diagnostic
lib.rs(105, 15): Actual error occurred here
```

Goldie Library:
https://github.com/rossmacarthur/goldie/blob/81daa5650aba992782b5d752c8839890bdd246d9/src/lib.rs#L104

```
/// Assert the JSON golden file matches.
#[macro_export]
macro_rules! assert_json {
    ($actual:expr) => {{
        let g = $crate::_new_goldie!();
        if let Err(err) = g.assert_json($actual) {
            ::std::panic!("{}", err);
        }
    }};
}
```

I asked in the Discord channel if there are any other libraries or substitutes for a golden test.

Reply:
Yes, because this is a `no_std` crate. You can read about it in Rust docs. But here we're importing it in tests, so it should be added to dev dependencies and not dependencies (see axelar-amplifier repo for reference)